libsplinter/src/service/scabbard/client/mod.rs

```diff
@@ -0,0 +1,306 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod contract_registry;
+mod error;
+mod execute;
+mod key;
+mod namespace;
+mod smart_permission;
+mod submit;
+mod transaction;
+mod upload;
+
+use std::fs::File;
+use std::io::{BufReader, Read};
+
+use sabre_sdk::protocol::payload::SabrePayload;
+pub use sawtooth_sdk::signing::secp256k1::Secp256k1PrivateKey;
+
+pub use error::Error;
+use submit::{submit_batch_list, wait_for_batches};
+use transaction::batch_payloads;
+pub use upload::{SmartContract, SmartContractMetadata};
+
+/// A client that can be used to submit transactions to a scabbard service.
+pub struct ScabbardClient {
+    url: String,
+    signing_key: Secp256k1PrivateKey,
+    unsent_payloads: Vec<SabrePayload>,
+}
+
+impl ScabbardClient {
+    /// Create a new `ScabbardClient` with the given `url` and a signing key loaded from the given
+    /// `key_file`.
+    pub fn new_with_local_signing_key(url: &str, key_file: &str) -> Result<Self, Error> {
+        Ok(Self::new(url, Self::load_signing_key_from_file(key_file)?))
+    }
+
+    /// Create a new `ScabbardClient` with the given `url` and `signing_key`.
+    pub fn new(url: &str, signing_key: Secp256k1PrivateKey) -> Self {
```

Comment:
this should take a Signer, not a key. Signer is not algorithm-specific, which will push the algorithm-specific code above this client library.

Reply:
Removed signing from this PR